bundles/org.eclipse.ui.editors/src/org/eclipse/ui/internal/texteditor/stickyscroll/StickyScrollingControl.java

```diff
@@ -394,6 +394,9 @@ private void addSourceViewerListeners() {
 		if (sourceViewer instanceof ITextViewerExtension4 extension) {
 			textPresentationListener = e -> {
 				Display.getDefault().asyncExec(() -> {
+					if (areStickyLinesOutDated(textWidget)) {
```

Comment:
Note, using asyncExec without checking if the widget is still alive is very common bug pattern. Could you please add a check that `textWidget` is not disposed? Otherwise it will fail in `textWidget.getLineCount()`

Reply:
Done, thanks for the hint.